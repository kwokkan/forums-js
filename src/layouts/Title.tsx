import Head from "next/head";

interface IProps {
    title?: string;
}

export function Title(props: IProps) {
    let title = "Forums JS";

    if (props.title) {
        title = `${props.title} - ${title}`;
    }

    return (
        <Head>
            <title>{title}</title>
            <link key="favicon" rel="icon" type="image/png" href="/logo.png" />
            <link key="rsuite-css" rel="stylesheet" href="https://cdn.jsdelivr.net/npm/rsuite@4.8.0/dist/styles/rsuite-default.min.css" integrity="sha256-iJOcspl2lqiHR4OrS1jeweVu820CH4rB1hjPnK0OOxA=" crossOrigin="anonymous" />
        </Head>
    );
}
