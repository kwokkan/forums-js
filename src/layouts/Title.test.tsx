import React from "react";
import { create } from "react-test-renderer";
import { Title } from "./Title";

test("Renders without props", () => {
    const tree = create(
        <Title />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

test("Renders with title", () => {
    const tree = create(
        <Title title="From test" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
