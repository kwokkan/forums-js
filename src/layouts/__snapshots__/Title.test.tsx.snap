// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders with title 1`] = `null`;

exports[`Renders without props 1`] = `null`;
